### PR TITLE
Application requirements.

### PR DESCRIPTION
**Pull request by dmdashenkov**
Added the very first variant of the requirements to the __Dashboard__ application.
These requirements may later migrate to `README` as soon as they are implemented.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

File: REQUIREMENTS.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+The __Spine Dashboard__ application is a command line tool for easy surfing through the content of 
+the __Spine__ storages. The main purpose of that is to give the users of the framework some basic 
+understanding about what is actually happening to their data.
+
+## Representation
+
+The app provides a human-friendly representation of:
+
+ - Read-side state (Projections) _[1]_
+ - Aggregate state (last Snapshots, Stand integration) _[1]_
```
Comment (armiol) [inline REQUIREMENTS.md]: Please fix the grammar.

File: REQUIREMENTS.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ - The time of the Projection Repository catch-up (last handled event time) _[0]_
+ - Live subscription to the Events and Failures _[2]_
+ - Live subscription to the Entity State changes _[2]_
+ 
+The [number] here represent the priority for implementing the feature, where 0 is the highest 
+priority.
+
+## Navigation
+
+The navigation within the app is represented via the main menu for selecting the type of the item to
```
Comment (armiol) [inline REQUIREMENTS.md]: Please rephrase. The navigation cannot be represented via the main menu.

File: REQUIREMENTS.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+view.
+
+
+
+After the type is selected, a table representing the first _N_ records is displayed. As the user
```
Comment (armiol) [inline REQUIREMENTS.md]: After the record type is selected, the first N records are displayed.

File: REQUIREMENTS.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+The navigation within the app is represented via the main menu for selecting the type of the item to
+view.
+
+
+
+After the type is selected, a table representing the first _N_ records is displayed. As the user 
+scrolls down, more records become visible.
+
+The table is scrollable via `Up`/`Down` keys.
```
Comment (armiol) [inline REQUIREMENTS.md]: I think we should mention `Search` functionality.

File: REQUIREMENTS.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+## GENERAL NOTE
+
+*This file contains a brief description of the __Dashboard__ application.*
+
+The __Spine Dashboard__ application is a command line tool for easy surfing through the content of
```
Comment (armiol) [inline REQUIREMENTS.md]: It's not _only_ for surfing through the storages. It's a monitoring app.

File: REQUIREMENTS.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
+## GENERAL NOTE
+
+*This file contains a brief description of the __Dashboard__ application.*
+
+The __Spine Dashboard__ application is a command line tool for easy surfing through the content of 
+the __Spine__ storages. The main purpose of that is to give the users of the framework some basic 
+understanding about what is actually happening to their data.
```
Comment (armiol) [inline REQUIREMENTS.md]: The target audience is those engineers which use Spine to build their applications.
The Dashboard is aimed to answer the following question quickly: what is going on under the hood right now.

File: REQUIREMENTS.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+The table is scrollable via `Up`/`Down` keys.
+
+In the table it is possible to search for a record by ID of by any other property. This operation 
+should not make any calls to the storage.
+
+## Storages
+
+Both __JDBC__ and __GAE-Java__ storage implementations are supported.
+
+If required it's possible to connect to a DB in a secure way (e.g. a firewall or credentials).
```
Comment (armiol) [inline REQUIREMENTS.md]: `in a secure way (e.g. a firewall or credentials).` — the `firewall` part does not make much sense.

File: REQUIREMENTS.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+## Usability
+
+Main requirement to the app usability is providing an easy way to start the data surfing (e.g. one
```
Comment (armiol) [inline REQUIREMENTS.md]: Please DRY.

File: REQUIREMENTS.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+Both __JDBC__ and __GAE-Java__ storage implementations are supported.
+
+If required it's possible to connect to a DB in a secure way (e.g. a firewall or credentials).
+
+## Usability
+
+Main requirement to the app usability is providing an easy way to start the data surfing (e.g. one 
+terminal command).
+
+It's possible to require a config file (e.g. `.spine.yml`) from the user to start using the app for
```
Comment (armiol) [inline REQUIREMENTS.md]: This is a good idea.

**Comment by dmdashenkov** [comment]
Closing this now with a hope for a better third-party solution, such as Zipkin, Stackdriver, etc.

**Pull request closed without merging**